contribs/src/main/java/com/netflix/conductor/contribs/kafka/KafkaPublishTask.java

```diff
@@ -0,0 +1,273 @@
+	private static final String MISSING_KAFKA_TOPIC = "Missing Kafka topic. See documentation for KafkaTask for required input parameters";
+	private static final String MISSING_KAFKA_VALUE = "Missing Kafka value.  See documentation for KafkaTask for required input parameters";
+	private static final String FAILED_TO_INVOKE = "Failed to invoke kafka task due to: ";
+
+	private ObjectMapper om = objectMapper();
+	private Configuration config;
+	private String requestParameter;
+	KafkaProducerManager producerManager;
+
+	private static final Logger logger = LoggerFactory.getLogger(KafkaPublishTask.class);
+
+
+	@Inject
+	public KafkaPublishTask(Configuration config, KafkaProducerManager clientManager) {
+		super(NAME);
+		this.config = config;
+		this.requestParameter = REQUEST_PARAMETER_NAME;
+		this.producerManager = clientManager;
+		logger.info("KafkaTask initialized...");
+	}
+
+	private static ObjectMapper objectMapper() {
+		final ObjectMapper om = new ObjectMapper();
+		om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+		om.configure(DeserializationFeature.FAIL_ON_IGNORED_PROPERTIES, false);
+		om.configure(DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES, false);
+		om.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+		om.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+		return om;
+	}
+
+	@Override
+	public void start(Workflow workflow, Task task, WorkflowExecutor executor) {
+
+		long taskStartMillis = Instant.now().toEpochMilli();
+		task.setWorkerId(config.getServerId());
+		Object request = task.getInputData().get(requestParameter);
+
+		if (Objects.isNull(request)) {
+			markTaskAsFailed(task, MISSING_REQUEST);
+			return;
+		}
+
+		KafkaPublishTask.Input input = om.convertValue(request, KafkaPublishTask.Input.class);
+
+		if (StringUtils.isBlank(input.getBootStrapServers())) {
+			markTaskAsFailed(task, MISSING_BOOT_STRAP_SERVERS);
+			return;
+		}
+
+		if (StringUtils.isBlank(input.getTopic())) {
+			markTaskAsFailed(task, MISSING_KAFKA_TOPIC);
+			return;
+		}
+
+		if (Objects.isNull(input.getValue())) {
+			markTaskAsFailed(task, MISSING_KAFKA_VALUE);
+			return;
+		}
+
+		try {
+			Future<RecordMetadata> recordMetaDataFuture = kafkaPublish(input);
+			try {
+				recordMetaDataFuture.get();
+				task.setStatus(Task.Status.COMPLETED);
+				long timeTakenToCompleteTask = Instant.now().toEpochMilli() - taskStartMillis;
+				logger.info("Published message {}, Time taken {}", input, timeTakenToCompleteTask);
+
+			} catch (ExecutionException ec) {
+				logger.error("Failed to invoke kafka task - execution exception {}", ec);
+				markTaskAsFailed(task, FAILED_TO_INVOKE + ec.getMessage());
+			}
+		} catch (Exception e) {
+			logger.error(String.format("Failed to invoke kafka task for input {} - unknown exception: {}", input), e);
+			markTaskAsFailed(task, FAILED_TO_INVOKE + e.getMessage());
+		}
+	}
+
+	private void markTaskAsFailed(Task task, String missingBootStrapServers) {
```

Comment:
The second parameter name - reason for incompletion should be generic, as it has been used with other reasons.